views/admin_approvals.jade

```diff
@@ -2,24 +2,55 @@ extends layout
 
 block bodyScripts
     script(type='text/javascript').
+        function approveSubscription(action, item) {
+          $.ajax({
+            url: `/admin/approvals/${action}`,
+            type: "POST",
+            dataType: "json",
+            data: {
+              id: item.id,
+              app: item.application.id,
+              api: item.api.id
+            }
+          })
+            .done(function (data, status) {
+                const sorting = $("#approvalsGrid").jsGrid("getSorting");
+                $ ("#approvalsGrid").jsGrid("loadData").done(function() {
+                $("#approvalsGrid").jsGrid("sort", { field: sorting.field, order: sorting.order })
+              })
+            })
+            .fail(function (xhr, status) {
+              alert(`Failed to ${action} subscription`);
+            });
+        }
         $(document).ready(function(){
            $('#approvalsGrid').jsGrid({
              width: "100%", pageSize: 10, sorting: true, paging: true, filtering: true, autoload: true,
              controller: {
-              loadData: function (filter) {
-                  var d = $.Deferred();
-                  var data = !{approvals};
-                  if (isEmptyGridFilter(filter)) {
-                    d.resolve(data);
+                loadData: function (filter) {
+                    var d = $.Deferred();
+                    $.ajax({
+                      url: "/admin/approvals",
+                      data: filter,
+                      dataType: "json"
+                    }).done(function(result) {
+                      if (!!filter.sortField && !!filter.sortOrder) {
```

Comment:
Filter object can contain sortField and sortOrder. However, we should applyGridFIlter only to filter fields, not for sorting.